crates/red_knot_test/src/lib.rs

```diff
@@ -4,10 +4,11 @@ use red_knot_python_semantic::types::check_types;
 use ruff_db::files::system_path_to_file;
 use ruff_db::parsed::parsed_module;
 use ruff_db::system::{DbWithTestSystem, SystemPathBuf};
+use ruff_source_file::OneIndexed;
 use std::collections::BTreeMap;
 use std::path::PathBuf;
 
-type Failures = BTreeMap<SystemPathBuf, matcher::FailuresByLine>;
+type Failures = BTreeMap<AbsoluteLineNumberPath, matcher::FailuresByLine>;
```

Comment:
Nit: I'm inclined to use a regular `Vec` here and instead sort it before returning it in `run_test`. Vec's are just overall simpler data structure (e.g. for iterating). I then would add the `md_offset` to the `FailuresByLine` data structure (do we still need `line_number` or can we replace it with the `md_offset`?)

Reply:
If we remove the grouping by embedded-file path from the output altogether, as I suggested below, that also suggests simplifying this data structure.